hypershift-operator/ignitionserver/main.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"regexp"
+	"strings"
+	"syscall"
+)
+
+// We only match /ignition/$NodePoolName
+var ignPathPattern = regexp.MustCompile("^/ignition/[^/ ]*$")
+
+// TODO (alberto): Parameterize listening port and URL to proxy.
+var addr = "0.0.0.0:9090"
+
+// This is the simplest http server that enable us to satisfy
+// 1 - 1 relation between clusters and ign endpoints.
+// At the moment this acts as purely a proxy towards MCS clusterIP type Services that
+// are named machine-config-server-$nodePoolName by convention and
+// are created on demand by NodePools,
+// this might / might not change in the future.
+// TODO (alberto): Support only https.
+// TODO (alberto): Support only token authenticated requests.
+// TODO (alberto): Metrics.
+func main() {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		log.Printf("User Agent: %s. Requested: %s", r.Header.Get("User-Agent"), r.URL.Path)
+
+		if !ignPathPattern.MatchString(r.URL.Path) {
+			// No pattern matched; send 404 response.
+			log.Printf("Path not found: %s", r.URL.Path)
+			http.NotFound(w, r)
+			return
+		}
+
+		// Get the nodePool name to build the MCS Service targetURL
+		// by convention machine-config-server-$nodePoolName/config/master.
+		urlPathSplit := strings.SplitAfter(r.URL.Path, "/")
+		if len(urlPathSplit) != 3 {
+			// This can only happen if our regExp is not working as expected.
+			log.Printf("Unexpected URL path: %s", r.URL.Path)
+			http.Error(w, fmt.Sprintf("Bad request, path %q is not supported", r.URL.Path), http.StatusBadRequest)
+			return
+		}
+		nodePoolName := urlPathSplit[2]
+		targetURL := fmt.Sprintf("http://machine-config-server-%s/config/master", nodePoolName)
```

Comment:
Is this working because of the default resolver suffixes (e.g. `.svc`)?

Reply:
that's my understanding, any `Service` within a ns should resolvable through `serviceName` or `serviceName.Namespace` https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#namespaces-of-services